src/units/abilities.cpp

```diff
@@ -1302,6 +1299,13 @@ unit_ability_list attack_type::get_specials_and_abilities(const std::string& spe
 	return abil_list;
 }
 
+int attack_type::composite_value(const unit_ability_list& abil_list, int base_value) const
+{
+	unit_abilities::effect align_effect(abil_list, base_value, shared_from_this());
+	int composite_value = align_effect.get_composite_value();
```

Comment:
The name `align_effect` is a little confusing, because this isn't always about alignment.

Reply:
Is there any reason this whole function can't just be reduced to one line?

```cpp
return unit_abilities::effect(abil_list, base_value, shared_from_this()).get_composite_value();
```

Reply:
No, thank you, itis corrected and rebased